docs/consensus/registry.md

```diff
@@ -0,0 +1,155 @@
+to deregister an entity and such a transaction will fail._
+
+<!-- markdownlint-disable line-length -->
+[`NewDeregisterEntityTx`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/registry/api?tab=doc#NewDeregisterEntityTx
+<!-- markdownlint-enable line-length -->
+
+### Register Node
+
+Node registration enables a new node to be created. A new register node
+transaction can be generated using [`NewRegisterNodeTx`].
+
+**Method name:**
+
+```
+registry.RegisterNode
+```
+
+The body of a register entity transaction must be a [`MultiSignedNode`]
+structure, which is a [multi-signed envelope] containing a [`Node`] descriptor.
+The signer of the transaction MUST be the node identity key.
+
+The owning entity MUST have the given node identity public key whitelisted in
+the `Nodes` field in its [`Entity`] descriptor.
+
+The node descriptor structure MUST be signed by all of the following keys:
+
+* Node identity key.
+* Consensus key.
+* Committee TLS certificate key.
+* Committee P2P key.
+
+Registering a node may require sufficient stake in the owning entity's
+[escrow account].
+
+<!-- markdownlint-disable line-length -->
+[`NewRegisterNodeTx`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/registry/api?tab=doc#NewRegisterNodeTx
+[`MultiSignedNode`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/common/node?tab=doc#MultiSignedNode
+[`Node`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/common/node?tab=doc#Node
+[multi-signed envelope]: ../crypto.md#multi-signed-envelope
+<!-- markdownlint-enable line-length -->
+
+### Unfreeze Node
+
+Node unfreezing enables a previously frozen (e.g., due to slashing) node to be
+thawed so it can again be eligible for committee elections. A new unfreeze node
+transaction can be generated using [`NewUnfreezeNodeTx`].
+
+**Method name:**
+
+```
+registry.UnfreezeNode
+```
+
+**Body:**
+
+```golang
+type UnfreezeNode struct {
+    NodeID signature.PublicKey `json:"node_id"`
+}
+```
+
+**Fields:**
+
+* `node_id` specifies the node identifier of the node to thaw.
+
+The transaction signer MUST be the entity key that owns the node.
+
+Thawing a node requires that the node's freeze period has already passed.
+
+<!-- markdownlint-disable line-length -->
+[`NewUnfreezeNodeTx`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/registry/api?tab=doc#NewUnfreezeNodeTx
+<!-- markdownlint-enable line-length -->
+
+### Register Runtime
+
+Runtime registration enables a new runtime to be created. A new register
+runtime transaction can be generated using [`NewRegisterRuntimeTx`].
+
+**Method name:**
+
+```
+registry.RegisterRuntime
+```
+
+The body of a register runtime transaction must be a [`SignedRuntime`]
+structure, which is a [signed envelope] containing a [`Runtime`] descriptor.
+The signer of the transaction MUST be the owning entity key.
+
+Registering a runtime may require sufficient stake in the owning entity's
+[escrow account].
```

Comment:
Might be useful to point out the consensus parameter that defines this threshold.